deploy/kubernetes/rbac.yaml

```diff
@@ -0,0 +1,44 @@
+# This YAML file contains all RBAC objects that are necessary to run external
+# CSI provisioner.
+#
+# In production, each CSI driver deployment has to be customized:
+# - to avoid conflicts, use non-default namespace and different names
+#   for non-namespaced entities like the ClusterRole
+# - decide whether the deployment replicates the external CSI
+#   provisioner, in which case leadership election must be enabled;
+#   this influences the RBAC setup, see below
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: csi-driver-registrar
+  # replace with non-default namespace name
+  namespace: default
+
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: driver-registrar-runner
+rules:
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["get", "list", "watch", "create", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["get", "update", "patch"]
```

Comment:
I haven't tried it, but you are probably right.

Do we need two rbac.yaml files then (one for registrar with `--kubelet-plugin-path`, one without), or do we only document the mode with `--kubelet-plugin-path`?

Reply:
hmmm I guess we technically still need to support the old way until we deprecate and remove it, so 2 manifests may be needed for now.

Reply:
do we have an issue open to make `--kubelet-plugin-path` a required argument? We wanted to remove the `Node` permissions ASAP for security reasons so I think it should be fairly high priority to deprecate the "old way."

Reply:
I think it's mainly that we need to deprecate the old registration mechanism in Kubernetes. And then we can remove it after two releases.  I suppose we could remove it immediately from the sidecars if we say some X version of the sidecar does not support an older Kubernetes version

Reply:
How about this?

- a single .rbac file
- the "nodes" commented out with a comment explaining that this is only needed when running without `--kubelet-plugin-path`

This is similar to (but not quite the same) how the optional leadership feature is handled in external-attacher and external-provisioner: https://github.com/kubernetes-csi/external-attacher/blob/5b2ae23a80b5e727f7d4bd8aadef2be3f2c4817e/deploy/kubernetes/rbac.yaml#L54-L65

The difference is that the external-attacher/provisioner have an optional `Role`, which was needed because the permissions are restricted to the namespace. We could split out the "nodes" access into a separate `ClusterRole`, but as this is going to be phased out, I consider that overkill.